facebook/src/com/facebook/share/model/GameRequestContent.java

```diff
@@ -21,37 +21,101 @@
 package com.facebook.share.model;
 
 import android.os.Parcel;
+import android.os.Parcelable;
 
-import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 /**
  * Describes the content that will be displayed by the GameRequestDialog
  */
 public final class GameRequestContent implements ShareModel {
-    public enum ActionType {
-        SEND,
-        ASKFOR,
-        TURN,
+    public enum ActionType implements Parcelable {
+        SEND("send"),
+        ASKFOR("askfor"),
+        TURN("turn");
+
+        private String value;
+
+        ActionType(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        @SuppressWarnings("unused")
+        public static final Creator<ActionType> CREATOR = new Creator<ActionType>() {
+            public ActionType createFromParcel(Parcel in) {
+                return ActionType.values()[in.readInt()];
+            }
+
+            public ActionType[] newArray(int size) {
+                return new ActionType[size];
+            }
+        };
+
+        @Override
+        public int describeContents() {
+            return 0;
+        }
+
+        @Override
+        public void writeToParcel(Parcel parcel, int i) {
+            parcel.writeInt(ordinal());
+        }
     }
 
-    public enum Filters {
-        APP_USERS,
-        APP_NON_USERS,
+    public enum Filters implements Parcelable {
+        APP_USERS("app_users"),
+        APP_NON_USERS("app_non_users");
+
+        private String value;
+
+        Filters(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        @SuppressWarnings("unused")
+        public static final Creator<Filters> CREATOR = new Creator<Filters>() {
+            public Filters createFromParcel(Parcel in) {
+                return Filters.values()[in.readInt()];
+            }
+
+            public Filters[] newArray(int size) {
+                return new Filters[size];
+            }
+        };
+
+        @Override
+        public int describeContents() {
+            return 0;
+        }
+
+        @Override
+        public void writeToParcel(Parcel parcel, int i) {
+            parcel.writeInt(ordinal());
+        }
     }
 
     private final String message;
-    private final String to;
+    private final List<String> recipients;
```

Comment:
Nice call on adding recipients. This is a much nicer api that is far less error prone